src/v/cluster/rm_stm_types.h

```diff
@@ -56,7 +56,9 @@ enum class partition_transaction_status : int8_t {
     //
     // Partition is aware of the transaction but the client has
     // not produced data as a part of this transaction.
-    initialized = 3
+    initialized = 3,
+    committed = 4,
+    aborted = 5
```

Comment:
nit: this is pretty confusing. i understand ongoing as a super set of states that contain the intialized one i.e. `initialized` transaction is still `ongoing`

Reply:
At the implementation level..

initialized --> begin_tx()  
ongoing --> produced data
committed/aborted - as the name suggestes

so each state assumes it has crossed the previous state .. (Additional confusion here is that ongoing was assigned enum value=0 < initialized =1, so for historical reasons I had to retain that order ( I can flip the order in the code if that makes it any easier to read).. what else is confusing here?